File: smart-ui/smart-ui-utils/src/main/resources/static/js/utils/Md5Utils.ts
import Md5 from 'utils/Md5'
/**
 * MD5工具类
 */
export default class Md5Utils {

  /**
   * 进行MD5加密
   * @param value 需要加密的值
   * @param hashIterations 加密次数
   */
  public static md5 (value: string, hashIterations: number = 1): string {
    let hash = Md5.hashStr(value).toString()
    if (hashIterations > 1) {
      for (let i=1; i<hashIterations; i++) {
        hash = Md5.hashStr(hash).toString()
      }
    }
    return hash
  }
}

File: smart-commons/smart-common/src/main/kotlin/com/smart/common/utils/security/MD5Utils.kt
package com.smart.common.utils.security

import org.apache.commons.codec.digest.DigestUtils
import java.io.IOException
import java.io.InputStream

/**
 * MD5 工具类
 * @author ming
 * 2019/6/12 上午10:21
 */
object MD5Utils {

    /**
     * MD5加密
     */
    fun md5(value: String, hashIterations: Int = 1): String {
        var hash = DigestUtils.md5Hex(value)
        if (hashIterations > 1) {
            for (i in 1 until hashIterations) {
                hash = DigestUtils.md5Hex(value)
            }
        }
        return hash
    }

    /**
     * 进行MD5加密
     * @param inputStream
     * @return
     */
    @Throws(IOException::class)
    fun md5(inputStream: InputStream): String {
        return DigestUtils.md5Hex(inputStream)
    }
}